### PR TITLE
builtin: add pi built-in function

### PR DESCRIPTION
**Pull request by cosmtrek**
Add `PI()` built-in function

### CODE REVIEW

File: expression/builtin_math_test.go [hunk 1]
```diff
@@ -351,3 +353,13 @@ func (s *testEvaluatorSuite) TestSqrt(c *C) {
 		c.Assert(v, DeepEquals, t["Ret"][0], Commentf("arg:%v", t["Arg"]))
 	}
 }
+
+func (s *testEvaluatorSuite) TestPi(c *C) {
+	defer testleak.AfterTest(c)()
+	fc := funcs[ast.PI]
+	var argNull types.Datum
+	f, _ := fc.getFunction(datumsToConstants([]types.Datum{argNull}), s.ctx)
```
Comment (coocood) [inline expression/builtin_math_test.go]: You can just pass `nil` as the first argument to `fc.getFunction`.

**Comment by coocood** [comment]
LGTM

**Comment by zimulala** [comment]
LGTM

**Comment by zimulala** [comment]
Please fix conflicting.

**Comment by cosmtrek** [comment]
@zimulala fixed conflicts

File: plan/typeinferer.go [hunk 2]
```diff
@@ -342,7 +342,9 @@ func (v *typeInferrer) handleFuncCallExpr(x *ast.FuncCallExpr) {
 		} else {
 			tp = types.NewFieldType(mysql.TypeLonglong)
 		}
-	case "ln", "log", "log2", "log10", "sqrt", "pi":
+	case "ln", "log", "log2", "log10", "sqrt", "pi", ast.Exp:
```
Comment (XuHuaiyu) [inline plan/typeinferer.go]: s/ast.Exp/"exp"

**Pull request merged**